lib/client.js

```diff
@@ -902,9 +922,17 @@ Request.prototype.end = function(fn){
   // set header fields
   for (var field in this.header) {
     if (null == this.header[field]) continue;
+    // blob and arraybuffer requests use responseType, not header Content-Type
+    // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Sending_and_Receiving_Binary_Data    
+    if (field === 'Content-Type' && this.responseType) continue;
```

Comment:
I'm definitely not familiar with this aspect of XMLHttpRequest, so I could be way off base here, but this line seems dubious. The `Content-Type` header is describing the body format for the outgoing request and the responseType is indicating how the response body is to be represented to the application JS code. Can you explain how/why `Content-Type` and `responseType` need to interact like this?